src/pages/vault/tutorials/[collectionSlug].tsx

```diff
@@ -10,10 +11,21 @@ import {
 
 export async function getStaticProps({
   params,
-}): Promise<{ props: CollectionPageProps }> {
+}): Promise<GetStaticPropsResult<CollectionPageProps>> {
   const { collectionSlug } = params
   const product = vaultData as LearnProductData
-  return await getCollectionPageProps(product, collectionSlug)
+
+  const props = await getCollectionPageProps(product, collectionSlug)
+
+  // If the collection doesn't exist, hit the 404
+  if (!props) {
```

Comment:
There's some repetition here, it could be lifted up into the server file but for some reason I like this being in the actual page file, since it so clearly shows how 404s are handled.